src/encoder/encoderfdkaac.cpp

```diff
@@ -0,0 +1,429 @@
+#include "encoder/encoderfdkaac.h"
+
+#include <QDir>
+#include <QStandardPaths>
+#include <QString>
+#include <QStringList>
+
+#include "engine/sidechain/enginesidechain.h"
+#include "recording/defs_recording.h"
+#include "util/logger.h"
+#include "util/sample.h"
+
+namespace {
+// recommended in encoder documentation, section 2.4.1
+const int kOutBufferBits = 6144;
+const mixxx::Logger kLogger("EncoderFdkAac");
+} // namespace
+
+EncoderFdkAac::EncoderFdkAac(EncoderCallback* pCallback)
+        : aacEncOpen(nullptr),
+          aacEncClose(nullptr),
+          aacEncEncode(nullptr),
+          aacEncInfo(nullptr),
+          aacEncoder_SetParam(nullptr),
+          m_aacAot(AOT_AAC_LC),
+          m_bitrate(0),
+          m_channels(0),
+          m_samplerate(0),
+          m_pCallback(pCallback),
+          m_library(nullptr),
+          m_pInputFifo(nullptr),
+          m_pFifoChunkBuffer(nullptr),
+          m_readRequired(0),
+          m_aacEnc(),
+          m_pAacDataBuffer(nullptr),
+          m_aacInfo(),
+          m_hasSbr(false) {
+    // Load shared library
+    // Code import from encodermp3.cpp
+    QStringList libnames;
+#ifdef __LINUX__
+    libnames << "fdk-aac";
+    libnames << "libfdk-aac.so.2";
+    libnames << "libfdk-aac.so.1";
+#elif __WINDOWS__
+    // Give top priority to libfdk-aac copied
+    // into Mixxx's installation folder
+    libnames << "libfdk-aac-2.dll";
+    libnames << "libfdk-aac-1.dll";
+
+    // Fallback and user-friendly method: use libfdk-aac
+    // provided with B.U.T.T installed in
+    // a standard location
+    QString buttFdkAacPath = buttWindowsFdkAac();
+    if (!buttFdkAacPath.isNull()) {
+        kLogger.debug() << "Found libfdk-aac at" << buttFdkAacPath;
+        libnames << buttFdkAacPath;
+    }
+
+    // Last resort choices: try versions with unusual names
+    libnames << "libfdk-aac.dll";
+    libnames << "libfdkaac.dll";
+#elif __APPLE__
+    // Using Homebrew ('brew install fdk-aac' command):
+    libnames << "/usr/local/lib/libfdk-aac.dylib";
+    libnames << "/usr/local/lib/libfdk-aac.2.dylib";
+    libnames << "/usr/local/lib/libfdk-aac.1.dylib";
+    // Using MacPorts ('sudo port install libfdk-aac' command):
+    libnames << "/opt/local/lib/libfdk-aac.dylib";
+    libnames << "/opt/local/lib/libfdk-aac.2.dylib";
+    libnames << "/opt/local/lib/libfdk-aac.1.dylib";
+#endif
```

Comment:
I think this code can be simplified by omitting the platform-specific file extensions:
https://doc.qt.io/qt-5/qlibrary.html#details
> If the file cannot be found, QLibrary tries the name with different platform-specific file prefixes, like "lib" on Unix and Mac, and suffixes, like ".so" on Unix, ".dylib" on the Mac, or ".dll" on Windows.

Reply:
Giving here the full path we bypass Qt's and Apples look up code, which is desired here. 
This also makes the debug output more verbose, because Qt is not able to return the full path of the library. 
It echoes only the input. 

The recommendation is valid to avoid target specific code on our side, but we have it anyway. 
So lets stick with the more verbose and faster version.

Reply:
> Giving here the full path we bypass Qt's and Apples look up code, which is desired here.

Why is that desired? I don't think we should bypass the usual behavior.